### PR TITLE
Fixing the Accepting of invitation endpoint

### PR DESCRIPTION
**Pull request by DoctorHeisenberg**
Done!

### CODE REVIEW

File: Provis.Core/Services/WorkspaceService.cs [hunk 1]
```diff
@@ -157,8 +157,8 @@ public async Task DenyInviteAsync(int id, string userid)
                 throw new HttpException(System.Net.HttpStatusCode.BadRequest, "You cannot deny this invite");
             }
 
-            if(inviteUserRec.IsConfirm==null)
-            inviteUserRec.IsConfirm = false;
+            if (inviteUserRec.IsConfirm == null)
+                inviteUserRec.IsConfirm = false;
```
Comment (YevheniiVovk) [inline Provis.Core/Services/WorkspaceService.cs]: Should we use null propagation ?

Comment (DoctorHeisenberg) [reply Provis.Core/Services/WorkspaceService.cs:161]: Thats not my piece of code, i just accidentally move the string. I will transfer the information to developer

File: Provis.Core/Services/WorkspaceService.cs [hunk 2]
```diff
@@ -199,11 +199,25 @@ public async Task AcceptInviteAsync(int id, string userid)
 
             if (inviteUserRec.ToUserId != userid)
             {
-                throw new HttpException(System.Net.HttpStatusCode.BadRequest, "You can't accept this invite");
+                throw new HttpException(System.Net.HttpStatusCode.BadRequest, "That's not yours invite!");
+            }
+
+            if (inviteUserRec.IsConfirm == true)
```
Comment (YevheniiVovk) [inline Provis.Core/Services/WorkspaceService.cs:205]: if (inviteUserRec.IsConfirm)

File: Provis.Core/Services/WorkspaceService.cs [hunk 3]
```diff
@@ -199,11 +199,25 @@ public async Task AcceptInviteAsync(int id, string userid)
...
+            UserWorkspace userWorkspace = new()
+            {
+                UserId = user.Id,
+                WorkspaceId = inviteUserRec.WorkspaceId,
+                RoleId = WorkSpaceRoles.MemberId
+            };
+
+            await _userWorkspaceRepository.AddAsync(userWorkspace);
+            await _userWorkspaceRepository.SaveChangesAsync();
             await _inviteUserRepository.SaveChangesAsync();
```
Comment (YevheniiVovk) [inline Provis.Core/Services/WorkspaceService.cs:220]: Two invokes of savechanges for a what reason ?

File: Provis.Core/Services/WorkspaceService.cs (hunk 2 above)
Comment (DoctorHeisenberg) [reply Provis.Core/Services/WorkspaceService.cs:205]: Не удается неявно преобразовать тип "bool?" в "bool". Существует явное преобразование (возможно, пропущено приведение типов).

Comment (DoctorHeisenberg) [reply Provis.Core/Services/WorkspaceService.cs:205]: Це nullable bool. Може я залишу як є?

Comment (YevheniiVovk) [reply Provis.Core/Services/WorkspaceService.cs:205]: why this field is nullable 
When we create a new invite the IsConfirm field should be defaulted to false

Comment (DoctorHeisenberg) [reply Provis.Core/Services/WorkspaceService.cs:205]: It is our logic. While IsConfirm is null - the user see the notification. When it become false - the notification is disappears

**Review by YevheniiVovk — APPROVED** [review]

**Review by AndriyKovalets — APPROVED** [review]

**Pull request merged**